### DISCUSSION TITLE
Activate alpinejs component in a popup only when the button to the popup is clicked

### DISCUSSION THREAD

**Discussion by anjanesh** [alpinejs/alpine#3694] [1. Help]
I have a popup form in TailWindCSS with vanilla JavaScript + AlpineJS.
I have `x-data` and `x-init` in my popup.
Thing is, this popup's alpinejs is initiated before even the popup is clicked.
So when I click the button for popup for the first-time it's fine but on closing the popup and re-opening the popup for the nth time, the data is not reset.
How do I deal with such a scenario ?

**Comment by ekwoka** [comment]
wrap it in an `x-if` so it completely unloads when not open.

**Reply by anjanesh** [reply]
You mean to say, to wrap the entire component in an `<template x-if="condition">` ?

**Reply by ekwoka** [reply]
That is one way to make it initialize fresh on every opening.
